compiler/z/codegen/FPTreeEvaluator.cpp

```diff
@@ -1268,24 +1225,31 @@ OMR::Z::TreeEvaluator::fbits2iEvaluator(TR::Node * node, TR::CodeGenerator * cg)
           generateLoadLiteralPoolAddress(cg, node, litBase);
           }
 
-       TR::MemoryReference * NaN              = generateS390MemoryReference((int32_t)0x7fc00000, TR::Int32, cg, litBase);
+       TR::MemoryReference* nanMemRef = nodeType == TR::Int64 ?
+                 generateS390MemoryReference(static_cast<int64_t>(0x7ff8000000000000), nodeType, cg, litBase) : 
+                 generateS390MemoryReference(static_cast<int32_t>(0x7fc00000),         nodeType, cg, litBase);
+       
+       auto testMnemonic = nodeType == TR::Int64 ? TR::InstOpCode::TCDB : TR::InstOpCode::TCEB;
 
        if (litBase)
           {
           deps->addPostCondition(litBase, TR::RealRegister::AssignAny);
           }
        deps->addPostCondition(targetReg, TR::RealRegister::AssignAny);
 
-       generateRRInstruction(cg, TR::InstOpCode::LDGR, node, targetFPR, targetReg);
+       generateRRInstruction(cg, TR::InstOpCode::LDGR, node, targetFPR, targetReg); 
 
-       generateRXEInstruction(cg, TR::InstOpCode::TCEB, node, targetFPR, generateS390MemoryReference(0x00F, cg), 0);
+       generateRXEInstruction(cg, testMnemonic, node, targetFPR, generateS390MemoryReference(0x00F, cg), 0);
 
        generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BZ, node, cleansedNumber);
 
        generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, cFlowRegionStart);
        cFlowRegionStart->setStartInternalControlFlow();
 
-       generateRXInstruction(cg, TR::InstOpCode::L, node, targetReg, NaN);
+       if (nodeType == TR::Int64)
+          targetReg = genericLoadHelper<64, 64, MemReg>(node, cg, NaN, NULL, false, true);
```

Comment:
This will fail to compile. The name of `NaN` has been changed. We'll need these changes accordingly.

Reply:
Yup. Already addressed here: 161c04a.

Once approved I'll do a rebase + squash to clean up the history.